domain/machine/service/service.go

```diff
@@ -70,3 +83,33 @@ func (s *Service) GetMachineLife(ctx context.Context, machineId string) (*life.L
 	life, err := s.st.GetMachineLife(ctx, machineId)
 	return life, errors.Annotatef(err, "getting life status for machine %q", machineId)
 }
+
+// ListAllMachines returns the ids of all machines in the model.
+func (s *Service) ListAllMachines(ctx context.Context) ([]string, error) {
+	machines, err := s.st.ListAllMachines(ctx)
+	if err != nil {
+		return nil, errors.Annotate(err, "retrieving all machines")
+	}
+	return machines, nil
+}
+
+// InstanceId returns the provider specific instance id for this machine.
+// If the machine is not provisioned, it returns a NotProvisionedError.
+func (s *Service) InstanceId(ctx context.Context, machineId string) (string, error) {
```

Comment:
In #17562 I introduced the concept of a machine id. Can we please swap over to using this?

I have made a mistake in the implementation by making it a UUID where it shouldn't be. I'll fix this up today.....

Reply:
I think the MachineID should traverse all the way down to the state layer as well.